doc/rst/technotes/libraries.rst

```diff
@@ -411,3 +419,83 @@ Exporting Symbols
 
 Only functions can be exported currently.  We hope to extend this support to
 types and global variables in the future.
+
+Arrays
+~~~~~~
+
+Arrays can be returned by exported Chapel functions as one of two C types:
+
+- ``chpl_external_array``
+
+  - For arrays that can be translated into native C or Python arrays.  In
+    Python, the contents of this type is copied into a Python array.
+
+- ``chpl_opaque_array``
+
+  - For arrays that are not currently translated.  In Python, this is used as a
+    field in a Python class named ``ChplOpaqueArray``.
+
+chpl_external_array
++++++++++++++++++++
+
+A ``chpl_external_array`` can be created in C or returned by a Chapel function
+declared as returning specific Chapel array types.  To create a
+``chpl_external_array`` in C, you can call:
+
+- ``chpl_make_external_array(elt_size, num_elts)`` to create an empty array of
+  the given size.
+
+- ``chpl_make_external_array_ptr(elts, num_elts)`` where ``elts`` is an existing
+  array of the given size.
+
+The memory owned by a ``chpl_external_array`` may or may not be considered to be
+the user's responsibility.  Users should call ``chpl_free_external_array`` to
```

Comment:
> The memory owned by a ``chpl_external_array`` may or may not be considered to be the user's responsibility.

That sentence may or may not be useful to readers...

But seriously I think it'd be better just leaving it out.